### ISSUE TITLE
SPIKE: Determine approach for required services as well as removing / archiving existing services

### ISSUE THREAD

**Issue by jilladams** [department-of-veterans-affairs/va.gov-cms#17929]
## User stories
**Required services**
Admin able to archive required service.
Admins able to remove a required service.
Admins should be able to unarchive a required service.

Other users not able to archive required service.
Other users not able to remove a required service.
Other users not able to unarchive a required service.

All users can tell if a service is required or not.
Users can see and edit required services ~, but not archive or remove them.~ (striking through this last clause as redundant of user stories above.)

**Removing services**
As a Vet Center editor, I want to remove a facility service from my facility.
As a product owner, I want to manage the number of nodes in the system, and archive nodes that are no longer useful. (Speaking to: "orphaned" facility service that might no longer be associated with a facility.)

## Considerations
Solution should be generalizable on Vet Centers today, and on VAMC / VBA potentially in the future.

## ACs
**Removing services**
- [x] Understand what "Remove" does today, in relation to Vet Centers & Vet Center facility services
   - [x] Understand the VAMC paradigm for "retiring" a VAMC Facility Service today

**Both**
- [x] Options are summarized with general level of effort for how to accomplish user stories
- [x] Summary is reviewed with Product & next steps are identified / ticketed

## Resources
[Work in progress matrix of functionality](https://docs.google.com/spreadsheets/d/1b7X3CfU15cA3gnVpmbRYohwWyNCCY9syRWSstYOqdy4/edit#gid=0)

### Removing services

Removing a Vet Center facility service via Inline Entity Form  does the following upon Vet Center node save:
- removes the facility reference on the service by unsetting the value of the `field_office`, thanks to a corresponding entity reference (CER).
- saves the service node, again via CER
- changes the name of the service node
- removes the facility name in the service name because there is no value in the `field_office` field
- results in a title such as " - Telehealth". 

The result is an orphan service with no apparent initially-associated facility, even though the facility can be inferred from the Section.

### "Retiring" a VAMC Facility Health Service by Archiving
The current workflow:
- An editor edits a VAMC Facility Health Service
- The editor sets the moderation state to Archived
- The editor saves the VAMC Facility Health Service

When the editor goes to the node:view of the facility itself, they can see the facility services and their moderation states. From that page they can also either view the service by clicking the linked name of the service or edit the service by clicking the Edit button.

The editor can bring the service out of "retirement" for "one last job" by: 
- Editing the VAMC Facility Health Service
-  Setting the moderation state to Published
- Saving the VAMC Facility Health Service

**Comment by jilladams** [comment]
Dave plans to collaborate with Christian as needed.

**Comment by omahane** [comment]
@jilladams @davidmpickett As we're looking to go away from IEF, I want to make sure I understand the future need for the following:
> As a product owner, I want to manage the number of nodes in the system, and archive nodes that are no longer useful. (Speaking to: "orphaned" facility service that might no longer be associated with a facility.)

If we go away from IEF, then we shouldn't have any orphaned services in the future because we will be using the archive process instead of the removal process for services no longer available at a Vet Center.

We can already see services that have been orphaned in [this view](https://prod.cms.va.gov/admin/content/audit?title_op=starts&title=-+&type_op=in&moderation_state_op=or&section=All).

Is there anything else we need to provide?

**Comment by davidmpickett** [comment]
Update, @omahane and I talked through his matrix on Friday and I will work on translating it into more product-focused documentation this week while he is out

**Comment by davidmpickett** [comment]
Update on my update. I have not had brain capacity to move this ticket forward this week

**Comment by davidmpickett** [comment]
# WIP proposal

**Comment by davidmpickett** [comment]
@omahane My proposal is ready for you to review before we share with product

**Comment by omahane** [comment]
@davidmpickett My first thought was about this box on Vet Center facility services node:edit. I added it to the issue where we are most likely to do the work:
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17676#issuecomment-2135633891

**Comment by omahane** [comment]
Another thought I had that might not be clear from this issue:
- By "All users can tell if a service is required or not," we are only speaking of those users who can access the node:edit of either the Vet Center facility or the Vet Center facility service.

**Comment by omahane** [comment]
@davidmpickett I made a slight change to the proposal, just to clarify that "All users can tell if a service is required or not" is covered by 2 tickets. I also added a [suggested design change](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17676#issuecomment-2135633891) on the facility service node:edit for required services. Otherwise, I think we covered it.

**Comment by davidmpickett** [comment]
> Another thought I had that might not be clear from this issue:
> 
> * By "All users can tell if a service is required or not," we are only speaking of those users who can access the node:edit of either the Vet Center facility or the Vet Center facility service.

Since this solution is generalizable, it would also mean VAMC editors for VAMC service and VBA editors for VBA services. We could reword it to "relevant users" or "Admin and non-admin users" since that's the main distinction we're drawing. 
There's other implicit user distinctions in the proposal. Or maybe just "users can" like 5) and 6) are worded

**Comment by omahane** [comment]
> > Another thought I had that might not be clear from this issue:
> > 
> > * By "All users can tell if a service is required or not," we are only speaking of those users who can access the node:edit of either the Vet Center facility or the Vet Center facility service.
> 
> Since this solution is generalizable, it would also mean VAMC editors for VAMC service and VBA editors for VBA services. We could reword it to "relevant users" or "Admin and non-admin users" since that's the main distinction we're drawing. There's other implicit user distinctions in the proposal. Or maybe just "users can" like 5) and 6) are worded

I mostly wanted to account for the scope of users with access to node:edit, rather than any user of the CMS. So far, we haven't tried to account for node:view, though, I suppose we could.

**Comment by davidmpickett** [comment]
@Agile6MSkinner @mmiddaugh 

Our proposal is ready for your review https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17929#issuecomment-2129913229

BLUF: There's only a couple stories that would actually need dedicated tickets (both of which are stubbed). Most of the other stories are already in place or would be handled simultaneously with another story. There's an optional enhancement that we didn't stub a ticket for.

**Comment by davidmpickett** [comment]
End of Sprint update:
Product folks [tagged for review on Slack](https://dsva.slack.com/archives/C0FQSS30V/p1716915787578549?thread_ts=1716584356.964639&cid=C0FQSS30V) & added to UX Sync agenda for Thursday

**Comment by mmiddaugh** [comment]
Thank you - the work is clear intended in #18003 and #17676 is clear

Do we have other issues to update the existing experience in the node (removing the table and connection to the service add/edit view?

Have we accounted for the experience for adding a new services? 
- required services should be automatically added to a new Vet Center and therefore not appear in the list available following selection of "Add a new service"
- editors understand which services are available to be added to their Vet Center and which are already associated 
- editors can see the moderation state of existing services (related: Vet Center services also have "In review" as a moderation state)

**Comment by davidmpickett** [comment]
> Thank you - the work is clear intended in #18003 and #17676 is clear
> 
> Do we have other issues to update the existing experience in the node (removing the table and connection to the service add/edit view?
> 

Covered by #18003 
 
> Have we accounted for the experience for adding a new services?
> 
> * required services should be automatically added to a new Vet Center and therefore not appear in the list available following selection of "Add a new service"
> * editors understand which services are available to be added to their Vet Center and which are already associated

Removing these from the dropdown would be handled by #17674

> * editors can see the moderation state of existing services (related: Vet Center services also have "In review" as a moderation state)

This will be part of the new view added in #18003

**Comment by davidmpickett** [comment]
Discussed in UX sync today https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17669#issuecomment-2136242290

Follow up tickets still need some refinement, but they have been identified and updated
